src/lib.rs

```diff
@@ -98,11 +89,9 @@ where
     /// Sets display [Orientation]
     ///
     pub fn set_orientation(&mut self, orientation: Orientation) -> Result<(), Error> {
-        let value = (self.madctl & 0b0001_1111) | orientation.value_u8();
-        self.write_command(Instruction::MADCTL)?;
-        self.write_data(&[value])?;
-        self.options.set_orientation(orientation);
-        self.madctl = value;
+        self.dcs
+            .write_command(self.madctl.orientation(orientation))?;
```

Comment:
I've clarified this a bit but I still think using `Madctl` as a value store is ok. If I were to introduce a new type it'd have to hold all the Madctl values inside as well and it seems like a needless code complication to me. The new version should be much cleaner to understand wrt. the value update on set_orientation.

Reply:
Hmm, sorry but I still don't like. It is very confusing that the `with_...` methods mutate `self` and also return a value. Why don't they just return a new `SetAddressMode`, instead of mutating the current one?

But IMO using three fields to `Display` is still the better option. Why do you think this would be needless coder complication? All that would be needed is to replace the `madctl` field with three fields for `color_order`, `orientation` and `refresh_order` and have a single `update_madctl` method, that builds the `SetAddressMode` mode command from these fileds.

Reply:
Sure but that's a lot of additional code complexity. Now we have 3 fields in Display instead of one, two of which need to just be set once so that Madctl gets recreated in `set_orientation`

As for `Madctl::with_...` those are there for the builder pattern for use in `init` of each model so you can chain things up instead of doing `value.set_...; value.set_...`. I've removed the "update in place" part now to make things more pure.

It just seems to me having a madctl copy that we keep updating is more straightforward. It's not super easy to understand but this isn't some extensible part of the library that model makes or other people who might have to chime in need to work with a lot.

Reply:
OK. The only remaining suggestion I have is to add a `must_use` attribute to the `with_...` methods.

Reply:
done